test/handlers/ErrorHandlerSpec.scala

```diff
@@ -0,0 +1,108 @@
+import play.api.mvc.{Headers, RequestHeader, Result, Results}
+import play.api.test.Helpers.*
+import uk.gov.hmrc.play.bootstrap.frontend.http.ApplicationException
+import scala.util.control.NoStackTrace
+
+import scala.concurrent.Future
+
+// scalastyle:off magic.number
+class ErrorHandlerSpec extends SpecBase with AppWithDefaultMockFixtures with Results {
+
+  private lazy val handler: ErrorHandler = app.injector.instanceOf[ErrorHandler]
+
+  "onClientError" - {
+    "must redirect to NotFound page when given a 404" in {
+
+      val result: Future[Result] = handler.onClientError(new FakeRequestHeader, 404)
+
+      status(result) mustBe SEE_OTHER
+      redirectLocation(result).value mustBe frontendAppConfig.notFoundUrl
+    }
+
+    "must redirect to BadRequest page when given a client error (400-499)" in {
+
+      forAll(Gen.choose(400, 499).suchThat(_ != 404)) {
+        clientErrorCode =>
+          beforeEach()
+
+          val result: Future[Result] = handler.onClientError(new FakeRequestHeader, clientErrorCode)
+
+          status(result) mustBe SEE_OTHER
+          redirectLocation(result).value mustBe s"${frontendAppConfig.departureHubUrl}/bad-request"
+      }
+    }
+
+    "must redirect to TechnicalDifficulties page when given any other error" in {
+
+      forAll(Gen.choose(500, 599)) {
+        serverErrorCode =>
+          beforeEach()
+
+          val result: Future[Result] = handler.onClientError(new FakeRequestHeader, serverErrorCode)
+
+          status(result) mustBe SEE_OTHER
+          redirectLocation(result).value mustBe frontendAppConfig.technicalDifficultiesUrl
+      }
+    }
+  }
+
+  "onServerError" - {
+    "return the result from ApplicationException" in {
+      val customResult                    = Ok("Custom error response")
+      val exception: ApplicationException = ApplicationException(customResult, "Test application exception")
+
+      val result: Future[Result] = handler.onServerError(fakeRequest, exception)
+
+      status(result) mustBe OK
+      contentAsString(result) mustBe "Custom error response"
+    }
+
+    "redirect to the internal server error page for an unhandled exception" in {
+      val exception = new RuntimeException("Unexpected error") with NoStackTrace
+
+      val result: Future[Result] = handler.onServerError(fakeRequest, exception)
+
+      status(result) mustBe SEE_OTHER
+      redirectLocation(result) mustBe Some(s"${frontendAppConfig.departureHubUrl}/internal-server-error")
+    }
+  }
+
+  class FakeRequestHeader extends RequestHeader {
+    override val target: RequestTarget = RequestTarget("/context/some-path", "/context/some-path", Map.empty)
+
+    override def method: String = "POST"
+
+    override def version: String = "HTTP/1.1"
+
+    override def headers: Headers = new Headers(Seq.empty)
+
+    override def connection: RemoteConnection = RemoteConnection("", secure = true, None)
+
+    override def attrs: TypedMap = TypedMap()
+  }
```

Comment:
This might be redundant. Does `fakeRequest` in SpecBase work as a valid RequestHeader?